ganttproject/src/net/sourceforge/ganttproject/export/ExporterToCSV.java

```diff
@@ -83,8 +131,7 @@ protected IStatus run() {
           // TODO Fix this ugly hack!! Ie make the settings available in a proper way
           GanttCSVExport exporter = new GanttCSVExport(getProject(),
               ((GanttProject) getProject()).getGanttOptions().getCSVOptions());
-          exporter.save(outputStream);
-          outputStream.flush();
+          exporter.save(outputStream, myFileTypeOption.proposeFileExtension());
```

Comment:
Notice that what you call "fileFormat" in the classes undeline is not actually a format -- it is file extension. This kinda smells. I suggest creating `enum SpreadsheetFormat` with format ids, use its values in `FileTypeOption` and pass enum value to `save`